mirgecom/gas_model.py

```diff
@@ -286,6 +277,19 @@ def species_diffusivity(self):
         return self.tv.species_diffusivity
 
 
+@dataclass_array_container
+@dataclass(frozen=True, eq=False)
+class PorousFlowFluidState(ViscousFluidState):
+    """Dependent variables for the (porous) fluid state.
+
+    .. attribute:: wv
+
+        Porous media flow state-dependent properties.
+    """
+
+    wv: PorousWallVars
+
+
```

Comment:
My main point for a major refactor is that we use `gas_model` everywhere and can't easily decouple totally fluid-only and solid+fluid because of that. Here, I follow the increasing complexity of inviscid -> viscous -> porous flow and I am happy with it.